src/components/ui/coaching-sessions/editor-cache-context.tsx

```diff
@@ -179,15 +190,41 @@ export const EditorCacheProvider: React.FC<EditorCacheProviderProps> = ({
             }
           });
 
-          setCache((prev) => ({
-            ...prev,
-            presenceState: {
-              ...prev.presenceState,
-              users: updatedUsers,
-              currentUser:
-                currentUserPresence || prev.presenceState.currentUser,
-            },
-          }));
+          // IMPORTANT: Preserve previous users who are no longer in states array
+          // as disconnected instead of removing them entirely.
+          // This ensures smooth UX when users go offline (they appear as disconnected
+          // rather than disappearing completely).
+          setCache((prev) => {
+            const mergedUsers = new Map(prev.presenceState.users);
+
+            // Mark users who disappeared from awareness as disconnected
+            for (const [userId, oldPresence] of prev.presenceState.users) {
+              if (!updatedUsers.has(userId) && oldPresence.status === 'connected') {
+                // User was connected but no longer in awareness states - mark as disconnected
+                mergedUsers.set(userId, {
+                  ...oldPresence,
+                  status: 'disconnected',
+                  isConnected: false,
+                  lastSeen: new Date()
+                });
+              }
```

Comment:
I'm curious what this provides? What is the issue if users "Disappear" and what does that mean exactly?